framework/db/src/main/java/com/cloud/utils/db/TransactionLegacy.java

```diff
@@ -1196,8 +1196,8 @@ private static String getDBHAParams(String dbName, Properties dbProps) {
         sb.append("failOverReadOnly=" + dbProps.getProperty("db." + dbName + ".failOverReadOnly"));
         sb.append("&").append("reconnectAtTxEnd=" + dbProps.getProperty("db." + dbName + ".reconnectAtTxEnd"));
         sb.append("&").append("autoReconnectForPools=" + dbProps.getProperty("db." + dbName + ".autoReconnectForPools"));
-        sb.append("&").append("secondsBeforeRetryMaster=" + dbProps.getProperty("db." + dbName + ".secondsBeforeRetryMaster"));
-        sb.append("&").append("queriesBeforeRetryMaster=" + dbProps.getProperty("db." + dbName + ".queriesBeforeRetryMaster"));
+        sb.append("&").append("secondsBeforeRetrySource=" + dbProps.getProperty("db." + dbName + ".secondsBeforeRetrySource"));
```

Comment:
are these options validated by MySQL? (that is do they work wrt the mysql connector or mysql server)?

Reply:
@rhtyd these options 'secondsBeforeRetrySource', 'queriesBeforeRetrySource' works with mysql connector.
Some references here:
- https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-config-failover.html
- https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html

 MySQL Connector 8.0 supports MySQL Server 5.6, 5.7 and  8.0.